test/types.spec.ts

```diff
@@ -99,3 +99,56 @@ test('runWith should allow void env', async ({runTSC}) => {
   expect(result.output).not.toContain('folio.config.ts(12');
   expect(result.output).not.toContain('folio.config.ts(13');
 });
+
+test('test.extend should check types', async ({runTSC}) => {
+  const result = await runTSC({
+    'folio.config.ts': `
+      export const test = folio.newTestType<{ foo: string }>();
+      class FooEnv {
+        beforeEach() {
+          return { foo: '17' };
+        }
+      }
+      const x: number = '123';  // To match line numbers easier.
+      export const test1 = test.extend({ beforeEach: ({ foo }) => { return { bar: parseInt(foo) + 42 }; } });
+      test.runWith({});  // error
+      test1.runWith({});  // error
+      test.runWith(new FooEnv());
+      test1.runWith({});  // error
+      test1.runWith(new FooEnv());
+      export const test2 = test1.extend({ beforeEach: ({ bar }) => { return { baz: bar - 5 }; } });
+      test2.runWith({});  // error
+      test2.runWith(new FooEnv());
+      export const test3 = test.extend({ beforeEach: ({ bar }) => { return { baz: bar - 5 }; } });  // error
+    `,
+    'a.spec.ts': `
+      import { test, test1, test2 } from './folio.config';
+      const x: number = '123';  // To match line numbers easier.
+      test('my test', async ({ foo }) => {});
+      test('my test', async ({ bar }) => {});  // error
+      test1('my test', async ({ foo, bar }) => {});
+      test2('my test', async ({ foo, bar, baz }) => {});
+      test2('my test', async ({ x }) => {}); // error
+    `
+  });
+  expect(result.exitCode).not.toBe(0);
```

Comment:
No need to change it if it works for you, but I think you can use `// @ts-expect-error` to make this a lot cleaner instead of grepping the output for each line.

Reply:
👍